File: Backend/DAL/Entities/MessageModel.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;
using DAL.Enums;

namespace DAL.Entities;

public record MessageModel
{
    [Key] public ulong MessageId { get; set; }
    public string DeviceId { get; set; }
    public string Recipient { get; set; }
    public string Content { get; set; }
    public MessageState State { get; set; }
    public long SentAt { get; set; }

    [JsonIgnore] public bool Updated { get; set; }
    [JsonIgnore] public string ConnectionId { get; set; }
    public long? ProceededAt { get; set; }
}

File: Frontend/Android/app/src/main/java/cz/stepes/smsgateway/models/MessageModel.kt
package cz.stepes.smsgateway.models

import com.beust.klaxon.Json

data class MessageModel(
    @Json(name = "messageId") val messageId: Long,
    @Json(name = "deviceId") val deviceId: String,
    @Json(name = "recipient") val recipient: String,
    @Json(name = "content") val content: String,
    @Json(name = "state") var state: Int,
    @Json(name = "sentAt") val sentAt: Long,
    @Json(name = "proceededAt") val proceededAt: Long?
)